Source/ARTConnection.m

```diff
@@ -303,3 +284,50 @@ + (instancetype)newWithConnectionEvent:(ARTRealtimeConnectionEvent)value {
 }
 
 @end
+
+@implementation ARTConnectionRecoveryKey
+
+- (NSString *)asJson{
+    NSError *error;
+    NSDictionary *object = @{
+        @"msgSerial": @(_msgSerial),
+        @"connectionKey": _connectionKey,
+        @"serials": _serials ? _serials : @{}
+    };
+    
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:object
+                                                       options:0
+                                                         error:&error];
+
+    if (!jsonData) {
+        NSLog(@"Got an error while creating JSON for recovery key: %@", error);
+    } else {
+        return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+    }
+    return nil;
+}
+
++(ARTConnectionRecoveryKey *)fromJson:(NSString *)json{
```

Comment:
I will change the type of json as it looks a better choice than a raw `NSString` but I'm not sure about passing an error pointer as a parameter here as there is no reason for caller to handle if there is an issue there, the method will simply return null

Reply:
Update on this: I changed the parameter to `id< ARTJsonCompatible>` but it doesn't play well with Swift types - String and optionals specifically. I don't really want to spend lots of time refactoring this as this is essentially an internal object and in fact it doesn't have to be json compatible, it just needs to be a string that can be encoded/decoded back.

Reply:
Agree, but `error:(NSError *_Nullable *_Nullable)errorPtr` should remain, I saw you use `@throw` there, but it's not used in this framework, except some questionable places (I mean usage is not systematic). I admit that it can be used in `jsonString` method though, because it's an impossible code path and using it at least makes some sense ("If it still happened somehow, well, throw it then").

Reply:
See https://github.com/ably/ably-cocoa/pull/1496#discussion_r1020065806